index.js

```diff
@@ -273,13 +273,16 @@ class GithubScm extends Scm {
      */
     _getCheckoutCommand(config) {
         const checkoutUrl = `${config.host}/${config.org}/${config.repo}`;
+        const sshCheckoutUrl = `git@${config.host}:${config.org}/${config.repo}`;
         const checkoutRef = config.prRef ? config.branch : config.sha; // if PR, use pipeline branch
         const command = [];
 
         // Git clone
         command.push(`echo Cloning ${checkoutUrl}, on branch ${config.branch}`);
-        command.push(`export SCM_URL=${checkoutUrl}`);
-        command.push('if [ ! -z $SCM_USERNAME ] && [ ! -z $SCM_ACCESS_TOKEN ]; then '
+        command.push(`if [ $SCM_CLONE_TYPE = "ssh" ]; then export SCM_URL=${sshCheckoutUrl}; `
+                    + `else export SCM_URL=${checkoutUrl}; fi`);
+        command.push('if [ ! $SCM_CLONE_TYPE = "ssh" ] && '
```

Comment:
@FenrirUnbound  Yeah so the `\"` will give problems. It will actually check if the variable has the `\"`

Anyway, I tested and without quotes should work fine.